src/runtime/rpc/rpc_host_port.cpp

```diff
@@ -86,6 +87,33 @@ host_port::host_port(rpc_address addr)
     _type = addr.type();
 }
 
+bool host_port::from_string(const std::string s)
+{
+    auto pos = s.find_last_of(':');
+    if (pos == std::string::npos) {
+        return false;
+    }
+    _host = s.substr(0, pos);
+    std::string port = s.substr(pos + 1);
+
+    if (!internal::buf2unsigned(port, _port) || _port > UINT16_MAX) {
```

Comment:
Consider add a new function `buf2uint16` ? In principle, `internal::buf2unsigned` is an internal function and should be used only in `src/utils/string_conv.h`.

On the other hand, there is no need to check the validity of `_port`, which has been done in `buf2unsigned`:
```suggestion
    if (!buf2uint16(port, _port)) {
```